Add agenda for 2024-09-27 (TPAC) (#681)
solid/specification: meetings/2024-09-27.md

# W3C Solid Community Group: W3C TPAC 2024

* Date: 2024-09-27T16:00:00Z
* Call: Instructions are restricted to W3C users. You need to [log in](https://auth.w3.org/?url=https%3A%2F%2Fwww.w3.org%2Fevents%2Fmeetings%2F6c6bbdef-5999-43f1-a75e-9cfcf8fc8379%2F) to see them.
* Chat: TPAC Zoom chat
* Repository: https://github.com/solid/specification
* Status: Published

## Present
* [Virginia Balseiro](https://virginiabalseiro.com/#me)
* [Sarven Capadisli](https://csarven.ca/#i)
* Laurens Debackere (Digitaal Vlaanderen)
* [Pierre-Antoine Champin](https://champin.net/#pa)
* [Anatoly Scherbakov](https://yeti.sh)
* [elf Pavlik](https://elf-pavlik.hackers4peace.net)
* [Rahul Gupta](https://cxres.pages.dev/profile#i)
* Wonsuk Lee (ETRI)
* Hadrian Zbarcea
* [Jesse Wright](https://www.cs.ox.ac.uk/people/jesse.wright/) (University of Oxford)
* Sehwan Jeon
* Youngmin Ji
* Hiroshi Ota

> ---

Reviewer: ```suggestion
* Sam Goto
* HK
* Hooman
* JK

---
```

* [Solid Code of Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md), [Positive Work Environment at W3C: Code of Conduct](https://www.w3.org/policies/code-of-conduct/)
* Operating principle for effective participation is to allow access across disabilities, across country borders, and across time. Feedback on tooling and meeting timing is welcome.
* If this is your first time, welcome! please introduce yourself.

### Scribes
* Sarven Capadisli

---

## Topics

### Overview

* VB: This meeting is intended to be an information session about the CG.
* VB: Meeting runs until 17:30 UTC (19:30 CET / 10:30 PDT). 10 minute break at 16:50 UTC (18:50 CET / 09:50 PDT).
* VB: Open agenda after the break. Proposals welcome at the bottom of the agenda.

### Introductions

* VB: Web dev, involved in Solid some time, am one of the co-chairs.
* PAC: INRIA and W3C. Involved in the CG sometime, and LWS charter, staff contact.
* Max: Alibaba Group and Alibaba Cloud, attending as an observer
* HK: Brazil Network info center. here as observer

> * Sam Goto : Google, also here as observer

Reviewer: ```suggestion
* SG: Google, also here as an observer.
```

> * WL(Wonsuk Lee): I am working for ETRI in South Korea. Started in project related to Solid. Here to contribute.

Reviewer: ```suggestion
* WL: I am working for ETRI in South Korea. Started in project related to Solid. Here to contribute.
```

* LD: One of the co-chairs of LWS. Here representing gov of flanders.
* SJ: SEAK. Attending as an observer 
* RG: From India. Been part of Solid CG since 2020. Been contributing to Solid Protocol and particularly done work on Notifications.

> * eP: Originally from Europe. lvingin in Mexio. Participated in Social WG and now in Solid CG, FedID CG. Also rejoined the Social CG.

Reviewer: ```suggestion
* eP: Originally from Europe. Living in in Mexico. Participated in Social WG and now in Solid CG, FedID CG. Also rejoined the Social CG.
```

* JW: Formerly at Inrupt. Now at U. of Oxford. Making decentralised agents that represent legal entities using SOlid-ish technologies.
* AS: My name's Anatoly Scherbakov, involved with JSON-LD working group
* SC: Similar history as eP, chaired this group for 4 years, worked on editing and authoring a bunch of the specs. I try to implement the specs to help me understand if it's working as intended. Looking forward to participating in LWS. 
* HZ: With Inrupt as solutions arhitect. Also co-chair of Solid CG.

> * JK: WebContacts Guidelines. Cognitive Accessibility TaskForce. Longtime W3C member.

Reviewer: ```suggestion
* JK: Web Content Accessibility Guidelines. Cognitive Accessibility Task Force. Long time W3C member.
```

* PAC: The other inputs are depends on the WG. Depends on the dependencies of the Protocol. At the momment it is still flexible. I suppose the CG could still work on those. In fact, the WG charter says it depends on the maturity of the works. IMO, the CG should refrain on the Protocl at the moment. And we can synch on regular basis. If CG wants to transfer other groups they can and of course if WG agrees.
* RG: When working on the new specification, are you going ot work on scratch or build on the current protocol. Because you're doing the Use Case from scratch. When new UCs emerge, it is not clear to me if you build on Solid Spec or start fresh.
* PAC: My understanding is that this will be the WG's definition.
* LD: See how input and output documents. There is the first meeting in two weeks time. It'd be premature to say that today.
* eP: For the CG-WG coordination. I posted FedID transition proposal: https://github.com/w3c-fedid/Administration/blob/main/proposals-CG-WG.md . In LWS there are tentative deliverables. We could try to adapt this process.

> * SC: I want to add to PAC's point. Regarding the protocol it's a clear document, an input document to be continued with LWS. Why any of these odcumnets make it to WG is because there is significant incubation, not only ??? but implementation experience. In the past CG has been trying to behave like a WG would do. Makes the transition easier. If you look at the charters, there's emphasis on that at some poitn to get to a rec there needs to be open and interoprable implementations to show that it's not just a document. If the CG can focus more on showing that these things are real, they're used, there's good adoption, thatwill help us strengthen the argument that the work should continue in the WG or elsewhere. I'd like to separate the concerns of not only polishing documents but carrying implementations and test suites forward. If we can make the case that we have a good track record with implementations it'd help push those forward. 

Reviewer: ```suggestion
* SC: I want to add to PAC's point. Regarding the protocol it's a clear document, an input document to be continued with LWS. Why any of these documents make it to WG is because there is significant incubation, not only writing a specification but implementation experience. In the past CG has been trying to behave like a WG would do. Makes the transition easier. If you look at the charters, there's emphasis on that at some point to get to a REC there needs to be open and interoperable implementations to show that it's not just a document. If the CG can focus more on showing that these things are real, they're used, there's good adoption, that will help us strengthen the argument that the work should continue in the WG or elsewhere. I'd like to separate the concerns of not only polishing documents but carrying implementations and test suites forward. If we can make the case that we have a good track record with implementations it'd help push those forward. 
```

* eP: re RG's points, I think it'd be good to extract the storage part and the N3 patch parts from the specs, and x,y,z, referenced. My suggestion would be to extrcating them to balance them. In the CG we have some issues on adjusting naming, e.g., resource server, and so on. So, before publishing FPWD, it'd be good to adjust them, and storage extracted into dedicated spec.
* LD: LWS Chair hat off: One of the htings we probably have to take into account is that we had two iterations of this charter, and some objectsion, e.g., solutions oriented vs. problem oriented. So, hesitant I'm hesitant to just refine those documents. That'd might be a difficult position.

### LWS WG
URL: https://www.w3.org/groups/wg/lws/

* VB: There is a [Call for Participation: Linked Web Storage Working Group Charter Approved; Join the LWS WG](https://lists.w3.org/Archives/Public/public-lws-wg/2024Sep/0000.html).
* LD: We're probably have our first meeting in two weeks. There is a doodle out now in the public-lws mailing list. I'd suggest folks to register for that meeting. We've two IEs; from SC and DZ. First meeting is probably just an intro meeting. Looking forward to getting started.

> * SC: Thanks for reviewing the request. Not sure how familiar PAC are with other activities happening in W3C regarding transitions, being more open with community involvement. There's a CG council, and there's been some activity in the past that came to my attention, trying to facilitate the continuation of the contributors. Not necessarily significant/any contributors. The idea is how can W3C improve that for the wider community. If people have been investing their time working CGs incubating the specs, how can they continue to get involved as a non-member. One of the venues is as an invited expert, but some of that rests on whether there's any asurances that they will be involved, what the criteria might be to get involve, involvement from the chairs, W3C team contacts, to look out for the contributors for the original specifications, test suites. Moving forward, looking out for contributors to the WG to be invited as an IE, and anyone who wishes to be an IE. There's issues and PRs for this against the charter drafts. Wondering if current LWS chairs are aware of those activities and initiatives and whetehr you are reaching iout to individuals to see if they'd like to join, what kind of criteria you're thinking of. 

Reviewer: ```suggestion
* SC: Thanks for reviewing the request. Not sure how familiar PAC are with other activities happening in W3C regarding transitions, being more open with community involvement. There's a CG council, and there's been some activity in the past that came to my attention, trying to facilitate the continuation of the contributors. Not necessarily significant/any contributors. The idea is how can W3C improve that for the wider community. If people have been investing their time working CGs incubating the specs, how can they continue to get involved as a non-member. One of the venues is as an invited expert, but some of that rests on whether there's any assurances that they will be involved, what the criteria might be to get involved as IE, active involvement from the chairs and W3C Team contacts to look out for the contributors for the original specifications, test suites - including QA documents. Moving forward, looking out for contributors to the WG to be invited as an IE. And, anyone who wishes to be an IE. There's issues and PRs for this against the charter drafts. Wondering if current LWS chairs are aware of those activities and initiatives and whether you are reaching out to individuals to see if they'd like to join, what kind of criteria you're thinking of. 
```

* LD: Support what PAC said. Ways of getting involved, as member, IE, or participating in discusisons or elsewhere. The spirit of openness is of course what we want.
* SC: Keeping it in sync with W3C activities is good as long as it's aligned. 

### Solid CG Updates and Demos at TPAC 2024
URL: https://www.w3.org/2024/09/TPAC/group-updates.html#solid

* VB: Thanks to eP, RG, SC for putting together the demos.
* eP: I don't know how many people saw them. If you have any questions if anything is unclear, comment on the video, Matrix, github discussion, questions, ect. Please engage.
* VB: eP published 3 parts. RG has on PREP. And SC ok dokieli.
* RG: Just +1 to what eP said.

> * SC: If people have questions a good place is dokieli github or mailing lists, chats. I'm happy to share more details on the screencast in opther meetings or special topic meetings. The screencast is intended to be short, no audio. If you want to get the whole picture you can watch in a loop, but you may miss some of the context. 

Reviewer: ```suggestion
* SC: If people have questions, a good place is [dokieli](https://dokie.li/) GitHub, or in the context of Solid, the public-solid mailing list and chats. I'm happy to share more details on the screencast in other meetings or special topic meetings. The screencast is intended to be short, no audio. Intended to be easily shareable to give a quick view. If you want to get the whole picture you can watch in a loop, but you may miss some of the context. 
```

* HZ: One of hte big issues was in that call was about pod portability. It is very hard to move data from one location to another. If links are given, they may be broken. I'm not sure if that's something we should do in the WG or the CG.
* eP: There was a meeting about portability is Social Web meeting. And there would be different proposal. One for custom domain names, anothr for different resolution mechanisms, e.g., that isn't based on dns. A matrix that requires 'those' and 'that' requirement. Also to see priorities. For me it is important, for every proposal, we can evaluate those UCs.
* RG: What eP seems to ask is for architectural decision records in parallel with UCs and specifications. Perhaps something think in the WG.

> * SC: CG have some issues on this. There's an ocean of issues on portability as HZ pointed out. One on privisioning a data storage. Some of that had some question marks. There's a proposal made by ??? submitted to CG but we did not do a good job evaluating that, having to do with redirects and deleting. Some of that can be evaluated in the CG still. There's work that already had been done and if it seems mature we can handover to WG anytime. In the CG we have plenty of threads/considerations put into our radar and we need to revisit instead of assuming it's not there. 

Reviewer: ```suggestion
* SC: CG have some issues on this already. There's an ocean of issues on portability as HZ pointed out. One on provisioning a data storage ( https://github.com/solid/specification/issues/317 ). Some of that had some question marks. For example, there's a proposal made by PDS Interop folks for [Solid Migrator](https://github.com/pdsinterop/solid-migrator) submitted to CG - by MdJ in https://github.com/solid/specification/issues/474 - but we did not do a good job evaluating that, having to do with redirects and deleting. See also https://github.com/solid/specification/issues/136 , https://github.com/solid/specification/issues/462 . Some of that can be evaluated in the CG still. There's work that already had been done and if it seems mature we can handover to WG anytime. In the CG we have plenty of threads/considerations put into our radar and we need to revisit instead of assuming it's not there. 
```

> * SC: I find this portability topic comes up a lot but an important distinction is what web architecture says about moving resources around, it's very clear about redirects. Another topic about moving data around. AFAIK those are separate things: moving URIs around than having a copy of your data in a new storage. The URI aspect is covered under webarch, copying/archiving your storage could be exporting a dump of your storage and having amechanism on the new storage to take that and extract. You have HTTP copy. Copying the data is different from saying this resource identified by this URI is moved to another location. The harder parts of the problem is coming up with URI persistance policies and enforcing. What kind of assurances will there be that they can continue toc use that when moved to another storage? Some of that is the policy work and not so much the architectural decision on how portocols and interfaces speak to each other. If we can find persistance policies that'd be the answer to the problem. The copying the data seems more complex technially but it's a lot simpler than handling the persistance issues. 

Reviewer: ```suggestion
* SC: I find this portability topic comes up a lot but an important distinction is what web architecture says about moving resources around, it's very clear about redirects. Another topic about moving data around. AFAIK those are separate things: moving URIs around than having a copy of your data in a new storage. The URI aspect is covered under webarch, copying/archiving your storage could be exporting a dump of your storage and having mechanism on the new storage to take that and extract, for example. There is even `HTTP COPY` (from WebDAV) and we had that in our radar ( https://github.com/solid/specification/issues/19 ). Copying the data is different from saying this resource identified by this URI is moved to another location. The harder parts of the problem is coming up with URI persistence policies and enforcing. What kind of assurances will there be that they can continue to use that when moved to another storage? Some of that is the policy work and not so much the architectural decision on how protocols and interfaces speak to each other. If we can find persistence policies that'd be the answer to the problem. The copying the data seems more complex technically but it's a lot simpler than handling the persistence issues. 
```

* eP: There's at least one Solid IdP using it. Few demo apps. Would that be enough or best way to communicate that for adoption in Solid.
* SG: Curious about the demo apps. Bigger and more is better than less and small. Feel and small is better than none (TM). I'm excited about Solid and IdP registration API. If enough people want it, the more people want it it'd be easier to make the case.
* eP: If LWS's authorization system, at least ??? would that help the FedCM's or only about current uses.
* SG: We're looking for real-world use as opposed to references and specs. But it doesn't hurt. The way in which the browser can be helpful to users.
* RG: SG, how is Solid perceived at Google, I mean broadly.
* SG: I can't speak on behalf of Google. Only a couple of promotions away from CEO. So, only Sam. Have worked years on SW. Have empathy. In Chrome, there are different levers. The work this group does, it is good for the web, in the same way the social web group is. Just personal views. You'll find a lot of empathy in Chrome for these things.

### AP with Solid or LWS or Storage or Whatever

> * SC: There's some initiative on Social Web circles to have Solid storages as activity stream collections, following contacts, sending messages. There's old issues for NSS. We have Activity Pods and other work trying to merge both works. Fold behind those works seem to be confused as to where those gaps in specs should be addressed: Solid specs, implementations, Social web? We should revisit some of these with Social Web folks. Solid CG could revisit. 

Reviewer: ```suggestion
* SC: There's some initiative on Social Web circles to have Solid storages as Activity Streams Collections, following contacts, sending messages. There's old issues for NSS originally requesting this (2017) - https://github.com/nodeSolidServer/node-solid-server/issues/621 . We have Activity Pods and other work trying to merge both works. Folks behind those works seem to be - rightfully - confused as to where those gaps in specs should be addressed: Solid specs, implementations, Social Web? We should revisit some of these with Social Web folks. Solid CG could revisit too. 
```

* HZ: One of hte big issues was in that call was about pod portability. It is very hard to move data from one location to another. If links are given, they may be broken. I'm not sure if that's something we should do in the WG or the CG.
* eP: There was a meeting about portability is Social Web meeting. And there would be different proposal. One for custom domain names, anothr for different resolution mechanisms, e.g., that isn't based on dns. A matrix that requires 'those' and 'that' requirement. Also to see priorities. For me it is important, for every proposal, we can evaluate those UCs.
* RG: What eP seems to ask is for architectural decision records in parallel with UCs and specifications. Perhaps something think in the WG.

> * SC: CG have some issues on this already. There's an ocean of issues. And, as HZ pointed out portability, we have issues related to that as well. And on provisioning a data storage ( https://github.com/solid/specification/issues/317 ). Some of that had some question marks. For example, there's a proposal made by PDS Interop folks for [Solid Migrator](https://github.com/pdsinterop/solid-migrator) submitted to CG - by MdJ in https://github.com/solid/specification/issues/474 , https://github.com/pdsinterop/solid-link-metadata - including implementation experience, but we did not do a good job evaluating that, having to do with redirects, deleting, moving. See also https://github.com/solid/specification/issues/136 , https://github.com/solid/specification/issues/462 . Some of that can be evaluated in the CG still. There's work that already had been done and if it seems mature we can handover to WG anytime. In the CG we have plenty of threads/considerations put into our radar and we need to revisit instead of assuming it's not there. 
> * SC: I find this portability topic comes up a lot but an important distinction is what web architecture says about moving resources around, it's very clear about redirects. Another topic about moving data around. AFAIK those are separate things: moving URIs around than having a copy of your data in a new storage. The URI aspect is covered under webarch, copying/archiving your storage could be exporting a dump of your storage and having mechanism on the new storage to take that and extract, for example. There is even `HTTP COPY` (from WebDAV) and we had that in our radar ( https://github.com/solid/specification/issues/19 ). Copying the data is different from saying this resource identified by this URI is moved to another location. The harder parts of the problem is coming up with URI persistence policies and enforcing. What kind of assurances will there be that they can continue to use that when moved to another storage? Some of that is the policy work and not so much the architectural decision on how protocols and interfaces speak to each other. If we can find persistence policies that'd be the answer to the problem. The copying the data seems more complex technically but it's a lot simpler than handling the persistence issues. 

Reviewer 2: ```suggestion
* SC: The CG has some issues on this already. There's an ocean of issues. And, as HZ pointed out — portability; we have issues related to that as well. And on provisioning a data storage <https://github.com/solid/specification/issues/317>. Some of that had some question marks. For example, there's a proposal made by PDS Interop folks for [Solid Migrator](https://github.com/pdsinterop/solid-migrator), submitted to CG by MdJ in <https://github.com/solid/specification/issues/474>, <https://github.com/pdsinterop/solid-link-metadata> — including implementation experience, but we did not do a good job evaluating that, having to do with redirects, deleting, moving. See also <https://github.com/solid/specification/issues/136>, <https://github.com/solid/specification/issues/462>. Some of that can still be evaluated in the CG. There's work that has already been done, and if it seems mature, we can hand it over to the WG, anytime. In the CG, we have plenty of threads/considerations put on our radar, and we need to revisit them instead of assuming it's not there. 
* SC: I find this portability topic comes up a lot, but an important distinction is what [Web Architecture](https://www.w3.org/TR/webarch/) says about moving _resources_ around — it's very clear about redirects. Moving _data_ around is another topic. AFAIK those are separate things: moving URIs around vs. having a copy of your data in a new storage. The URI aspect is covered under webarch; copying/archiving your storage could be exporting a dump of your storage and having mechanisms on the new storage to take that and extract, for example. There is even `HTTP COPY` (from WebDAV), and we had that on our radar <https://github.com/solid/specification/issues/19>. Copying the data is different from saying "this resource identified by this URI is moved to another location". The harder parts of the problem are coming up with URI persistence policies and then enforcing them. What kind of assurances will there be that they can continue to use that when moved to another storage? Some of that is policy work and not so much the architectural decision of how protocols and interfaces speak to each other. If we can find persistence policies, that'd be the answer to the problem. Copying the data seems more complex technically, but it's a lot simpler than handling the persistence issues. 
```

* eP: what I'd like to get out is a strategy for UCs. For me the UCs are real-world, as opposed to the solutions. Working with the Food Network, open source software. It'd be good to connect with existing communities with user bases. Easy to experiment with. Wha thtye do but also using Solid. We should be hte last peopel to come pu with UCs. Get UCs from facing the problems. We used the solid forum, which is open source also. social web also wants to to do threads. We can showcase how this UC can overlap with ??? If we get a UC, we extract requirements, and some may be out of scope. Since some of the rquirements may be outside of hte scope, ??? then some will be for the WG and others left for the CG. Real challenge is to get the UCs from out there.

> * RG: To answer HZ's UC quickly using SC's answer. Goign got flip SC's answer. I think this is not a problem for just Solid or LWS. Maybe for Web platform, the reality is 1B people in a country like India will not be abl to own a domain name. Portability is not always going ot work, and I don't tihnk we are in a place to answer it.

Reviewer 2: ```suggestion
* eP: What I'd like to get out is a strategy for UCs. For me, the UCs are real-world, as opposed to the solutions. Working with the Food Network, open source software. It'd be good to connect with existing communities with user bases. Easy to experiment with. What they do, but also using Solid. We should be the last people to come up with UCs. Get UCs from facing the problems. We used the Solid forum, which is also open source. Social web also wants to to do threads. We can showcase how this UC can overlap with ??? If we get a UC, we can extract requirements, and some may be out of scope. Since some of the requirements may be outside of the WG scope, ??? then some will be for the WG and others left for the CG. Real challenge is to get the UCs from out there.
* RG: To answer HZ's UC quickly, using SC's answer. Going got flip SC's answer. I think this is not a problem for just Solid or LWS. Maybe for Web platform, the reality is 1Billion people in a country like India will not be able to own a domain name. Portability is not always going to work, and I don't think we are in a place to answer it.
```

### AP with Solid or LWS or Storage or Whatever

* SC: There's some initiative on Social Web circles to have Solid storages as Activity Streams Collections, following contacts, sending messages. There's old issues for NSS originally requesting this (2017) - https://github.com/nodeSolidServer/node-solid-server/issues/621 . We have Activity Pods and other work trying to merge both works. Folks behind those works seem to be - rightfully - confused as to where those gaps in specs should be addressed: Solid specs, implementations, Social Web? We should revisit some of these with Social Web folks. Solid CG could revisit too. 

> * PAC: Had a long and interesting topics on those with EP. There is the Social Web Maintenance WG charters.. The channels are open and agree with SC. 

Reviewer 2: ```suggestion
* SC: There are some initiatives in Social Web circles to have Solid storages as Activity Streams Collections, following contacts, sending messages. There are old issues for NSS originally requesting this (circa 2017) - <https://github.com/nodeSolidServer/node-solid-server/issues/621>. We have Activity Pods and other work trying to merge both works. Folks behind those works seem to be — rightfully — confused as to where those gaps in specs should be addressed: Solid specs, implementations, Social Web? We should revisit some of these with Social Web folks. Solid CG could revisit too. 
* PAC: Had a long and interesting topics on those with EP. There is the Social Web Maintenance WG charter. The channels are open and agree with SC.
```
